cli/server.go

```diff
@@ -1512,3 +1517,64 @@ func redirectHTTPToAccessURL(handler http.Handler, accessURL *url.URL) http.Hand
 func isLocalhost(host string) bool {
 	return host == "localhost" || host == "127.0.0.1" || host == "::1"
 }
+
+func makeLogger(cmd *cobra.Command, cfg *codersdk.DeploymentConfig) (slog.Logger, func(), error) {
```

Comment:
We should add testing around this. If logging breaks in any way, we really hurt our customers.

Reply:
@coadler since you can't read sinks from an instantiated logger, you could make this return `([]slog.Sink, slog.Level, func() error)` instead and instantiate the logger from the caller.

Reply:
I like that!